File: nodetest/Utils/ElementUtils.cs
using System.Reflection;
using Microsoft.AspNetCore.Components;
using Microsoft.JSInterop;
using nodetest.Models;

namespace nodetest.Utils;

public static class ElementUtils
{
    public class BoundingClientRect
    {
        public Vector2 Center => new()
        {
            X = (int)((Left + Right) / 2),
            Y = (int)((Top + Bottom) / 2)
        };
        
        public double X { get; set; }
        public double Y { get; set; }
        public double Width { get; set; }
        public double Height { get; set; }
        public double Top { get; set; }
        public double Right { get; set; }
        public double Bottom { get; set; }
        public double Left { get; set; }
    }

    private static Func<ElementReference, IJSRuntime> _getJsRuntime;
    static ElementUtils()
    {
        var elementReferenceExtensionsType = typeof(ElementReferenceExtensions);
        
        var getJsRuntimeMethodInfo =  elementReferenceExtensionsType.GetMethod("GetJSRuntime", BindingFlags.NonPublic | BindingFlags.Static);

        _getJsRuntime = Delegate.CreateDelegate(typeof(Func<ElementReference, IJSRuntime>), getJsRuntimeMethodInfo) as Func<ElementReference, IJSRuntime>;
    }
    public static async Task<string> GetNameAsync(this ElementReference reference)
    {
        var runtime = _getJsRuntime(reference);
        
        return await runtime.InvokeAsync<string>("ElementUtils.getName", reference);
    }
    
    public static async Task<BoundingClientRect> GetBoundingClientRect(this ElementReference reference)
    {
        var runtime = _getJsRuntime(reference);
        
        return await runtime.InvokeAsync<BoundingClientRect>("ElementUtils.getBoundingClientRect", reference);
    }
}

File: nodetest/wwwroot/js/ElementUtils.js
var ElementUtils = ElementUtils || {
    getName(element) {
        console.log(element);
        return element.toString();
    },
    getBoundingClientRect(element) {
        return element.getBoundingClientRect();
    },
    getOffset(element) {
        const x = $(element).offset().left;
        const y = $(element).offset().top;
        return {x: Math.trunc(x), y: Math.trunc(y)}
    },
    
    getWidth(element) {
        return Math.trunc($(element).width());
    },
    getHeight(element) {
        return Math.trunc($(element).height());
    }


};